core/frontend/src/tile/map/MapLayerFormatRegistry.ts

```diff
@@ -144,18 +144,52 @@ export class MapLayerFormatRegistry {
   public setAccessClient(formatId: string, accessClient: MapLayerAccessClient): boolean {
     const entry = this._formats.get(formatId);
     if (entry !== undefined) {
-      entry.accessClient = accessClient;
+      entry.client = accessClient;
       return true;
     }
     return false;
   }
 
-  /** @beta */
+  /**
+  * Returns the active session client for a given format.
+   *  @beta */
   public getAccessClient(formatId: string): MapLayerAccessClient | undefined {
     if (formatId.length === 0)
       return undefined;
 
-    return this._formats.get(formatId)?.accessClient;
+    const accessClient = this._formats.get(formatId)?.client;
+    if (accessClient && typeof((accessClient as any).getAccessToken) === "function") {
```

Comment:
When would accessClient not be a function? Your type is `getAccessToken(params: MapLayerAccessTokenParams): Promise<MapLayerAccessToken | undefined>;
`

Reply:
`MapLayerFormatEntry.client`'s type if `MapLayerAccessClient|MapLayerSessionClient`.  Those 2 sub-types are interfaces, and we need to determine which of the types is being active it.  To achieve it, I check if a mandatory function is available.

So to answer your first question,  `MapLayerFormatEntry.client.getAccessToken` will either be `function` OR `undefined`

Reply:
Can't you check if it's undefined, then?